### DISCUSSION TITLE
getItem and setItem in storage persister with query key

### DISCUSSION THREAD

**Discussion by chshouyu** [TanStack/query#3791] [Ideas] [answered]
I noticed that the AsyncStorage interface is:

```
interface AsyncStorage {
  getItem: (key: string) => Promise<string>
  setItem: (key: string, value: string) => Promise<unknown>
  removeItem: (key: string) => Promise<unknown>
}
```

I found that persistor will create a single stoage object, if the cacheTime is set very long and the query data is very large, the cache 
 size will be huge. if use `setItem` to save cache to a file, the file size will be in the megabyte range, this maybe a issue when serialize data with JSON.stringify or deserialize data with JSON.parse.

So, I think if when can add query key in Storage set & get method, In this way, we can save and get cache data one by one, a query's cache data can be saved to one file or a record in sqlite, etc.

**Comment by TkDodo** [comment, marked as answer]
see: https://github.com/TanStack/query/discussions/2649
